build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
++  // |effective_document_url|. It is the caller's responsibility to calculate
++  // |effective_document_url| based on match_origin_as_fallback().
++  bool MatchesDocument(const GURL& effective_document_url,
++                       bool is_subframe) const;
++
++  // Serializes the UserScript into a pickle. The content of the scripts and
++  // paths to UserScript::Files will not be serialized!
++  void Pickle(base::Pickle* pickle) const;
++
++  // Deserializes the script from a pickle. Note that this always succeeds
++  // because presumably we were the one that pickled it, and we did it
++  // correctly.
++  void Unpickle(const base::Pickle& pickle, base::PickleIterator* iter);
++
++ private:
++  // base::Pickle helper functions used to pickle the individual types of
++  // components.
++  void PickleGlobs(base::Pickle* pickle,
++                   const std::vector<std::string>& globs) const;
++  void PickleHostID(base::Pickle* pickle, const HostID& host_id) const;
++  void PickleURLPatternSet(base::Pickle* pickle,
++                           const URLPatternSet& pattern_list) const;
++  void PickleScripts(base::Pickle* pickle, const FileList& scripts) const;
++
++  // Unpickle helper functions used to unpickle individual types of components.
++  void UnpickleGlobs(const base::Pickle& pickle,
++                     base::PickleIterator* iter,
++                     std::vector<std::string>* globs);
++  void UnpickleHostID(const base::Pickle& pickle,
++                      base::PickleIterator* iter,
++                      HostID* host_id);
++  void UnpickleURLPatternSet(const base::Pickle& pickle,
++                             base::PickleIterator* iter,
++                             URLPatternSet* pattern_list);
++  void UnpickleScripts(const base::Pickle& pickle,
++                       base::PickleIterator* iter,
++                       FileList* scripts);
++
++  // The location to run the script inside the document.
++  RunLocation run_location_ = DOCUMENT_IDLE;
++
++  // The namespace of the script. This is used by Greasemonkey in the same way
++  // as XML namespaces. Only used when parsing Greasemonkey-style scripts.
++  std::string name_space_;
++
++  // The script's name. Only used when parsing Greasemonkey-style scripts.
++  std::string name_;
++
++  // A longer description. Only used when parsing Greasemonkey-style scripts.
++  std::string description_;
++
++  // A version number of the script. Only used when parsing Greasemonkey-style
++  // scripts.
++  std::string version_;
++
++  // Greasemonkey-style globs that determine pages to inject the script into.
++  // These are only used with standalone scripts.
++  std::vector<std::string> globs_;
++  std::vector<std::string> exclude_globs_;
++
++  // URLPatterns that determine pages to inject the script into. These are
++  // only used with scripts that are part of extensions.
++  URLPatternSet url_set_;
++  URLPatternSet exclude_url_set_;
++
++  // Li
```

Comment:
I would suggest to enable this by default, at least until we remove the "experimental" naming

Reply:
it makes browsing slow, that's why I added it

Reply:
Let's mention it in the flag description then?

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c